### PR TITLE
Create separate image for Windows binary artifacts

### PR DESCRIPTION
**Pull request by yselkowitz**
This would have to be paired with a new container request and PR to ocp-build-data (which I can also provide).  How these binaries would be extracted and distributed to users would be up to ART.

CC: @dcbw

### CODE REVIEW

**Comment by yselkowitz** [comment]
/hold
still waiting to hear if this is the route you want to go for providing these or not.

**Comment by dcbw** [comment]
@yselkowitz it was our understanding that we could do something like https://github.com/openshift/ovn-kubernetes/pull/69 and ART could just pull the EXE out of the image? At least that was my understnading of the conversation last week...

**Comment by yselkowitz** [comment]
Do you really want to increase the size of ovn-kubernetes solely for the benefit of a Windows binary which is only present to be extracted and published elsewhere?  That's why I suggested a separate image for that purpose.

**Comment by dcbw** [comment]
@yselkowitz sure, a separate image would not have that problem. I have no idea what impact that has on ART's process though. It was clear to me that it was trivial for ART to extract the Windows binary from the existing image. But it's much less clear to me what all needs to be done for a new image.

If ART is fine with a new image then sure.

File: Dockerfile.artifacts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+WORKDIR /go-controller
+COPY go-controller/ .
+
+# build the binaries
+RUN CGO_ENABLED=0 make windows
+
+FROM openshift/origin-ovn-kubernetes
+
+COPY --from=builder /go-controller/_output/go/bin/windows/ovnkube.exe /usr/share/openshift/windows/
```
Comment (dcbw) [inline Dockerfile.artifacts]: @yselkowitz we don't need ovnkube.exe or ovn-k8s-cni-overlay.exe packaged. The only thing we need in the image is the `hybrid-overlay.exe`

File: go-controller/hack/build-go.sh [hunk 2]
```diff
@@ -35,11 +35,11 @@ build_windows_binaries() {
     set -x
     for bin in "$@"; do
         binbase=$(basename ${bin})
-        GOOS=windows go build -v \
+        GOOS=windows GOARCH=amd64 go build -v \
```
Comment (dcbw) [inline go-controller/hack/build-go.sh]: @yselkowitz These changes should go upstream

Comment (yselkowitz) [reply go-controller/hack/build-go.sh]: Filed: https://github.com/ovn-org/ovn-kubernetes/pull/995

Comment (yselkowitz) [reply go-controller/hack/build-go.sh]: Merged upstream.

File: Dockerfile.artifacts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# This Dockerfile builds an image containing the Windows ovn binaries
+# layered on top of the Linux ovn-kubernetes image.
+FROM openshift/origin-release:golang-1.12 AS builder
+
+WORKDIR /go-controller
+COPY go-controller/ .
+
+# build the binaries
+RUN CGO_ENABLED=0 make windows
```
Comment (trozet) [inline Dockerfile.artifacts]: @yselkowitz do you need to rebase and remove the windows building from #69 ?

**Comment by yselkowitz** [comment]
Since #69 has since been merged, the second half of this is moot.  Closing this and filing a separate PR for just the first commit (which is now upstream).

**Pull request closed without merging**